src/components/IstioWizards/WeightedRouting.tsx

```diff
@@ -0,0 +1,186 @@
+            prevState.workloads[nodeId[j]].weight = maxWeight - sumWeights;
+          }
+          sumWeights += prevState.workloads[nodeId[j]].weight;
+        }
+
+        // Adjust last element
+        if (nodeId.length > 0 && sumWeights < maxWeight) {
+          prevState.workloads[nodeId[nodeId.length - 1]].weight += maxWeight - sumWeights;
+        }
+
+        return {
+          workloads: prevState.workloads
+        };
+      },
+      () => this.props.onChange(this.checkTotalWeight(), this.state.workloads, false)
+    );
+  };
+
+  onLock = (workloadName: string, locked: boolean) => {
+    this.setState(prevState => {
+      let maxWeights = 100;
+      for (let i = 0; i < prevState.workloads.length; i++) {
+        if (prevState.workloads[i].name === workloadName) {
+          prevState.workloads[i].locked = locked;
+        }
+        // Calculate maxWeights from locked nodes
+        if (prevState.workloads[i].locked) {
+          maxWeights -= prevState.workloads[i].weight;
+        }
+      }
+      // Update non locked nodes maxWeight
+      for (let i = 0; i < prevState.workloads.length; i++) {
+        if (!prevState.workloads[i].locked) {
+          prevState.workloads[i].maxWeight = maxWeights;
+        }
+      }
+      return {
+        workloads: prevState.workloads
+      };
+    });
+  };
+
+  checkTotalWeight = (): boolean => {
+    // Check all weights are equal to 100
+    return this.state.workloads.map(w => w.weight).reduce((a, b) => a + b, 0) === 100;
+  };
+
+  render() {
+    const isValid = this.checkTotalWeight();
+    return (
+      <div>
+        <ListView>
+          {this.state.workloads.map((workload, id) => {
+            return (
+              <ListViewItem
+                key={'workload-' + id}
+                leftContent={<ListViewIcon type={wkIconType} name={wkIconName} />}
+                heading={workload.name}
+                description={
+                  <Slider
+                    id={'slider-' + workload.name}
+                    key={'slider-' + workload.name}
+                    tooltip={true}
+                    input={true}
+                    inputFormat="%"
+                    label={'Traffic Weight'}
+                    value={workload.weight}
+                    min={0}
+                    max={workload.maxWeight}
+                    onSlide={value => {
+                      this.onWeight(workload.name, value as number);
+                    }}
+                    locked={this.state.workloads.length > 1 ? workload.locked : true}
+                    showLock={this.state.workloads.length > 1}
+                    onLock={locked => this.onLock(workload.name, locked)}
+                  />
+                }
+              />
+            );
+          })}
+        </ListView>
+        {!isValid && <div className={validationStyle}>The sum of all weights must be 100 %</div>}
+      </div>
+    );
```

Comment:
Replace 
```html
<div></div>
```
 with 
```html
<></>
```
 to avoid depth dom

Reply:
+1